packages/splits-sdk/src/client/warehouse.ts

```diff
@@ -0,0 +1,1438 @@
+      functionName: 'transfer',
+      functionArgs: [receiver, fromHex(token, 'bigint'), amount],
+      transactionOverrides,
+    })
+
+    return result
+  }
+
+  protected async _transferFrom({
+    sender,
+    receiver,
+    token,
+    amount,
+    transactionOverrides = {},
+  }: WarehouseTransferFromConfig): Promise<TransactionFormat> {
+    validateAddress(sender)
+    validateAddress(receiver)
+    validateAddress(token)
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    const result = await this._executeContractFunction({
+      contractAddress: this._warehouseContract.address,
+      contractAbi: warehouseAbi,
+      functionName: 'transferFrom',
+      functionArgs: [sender, receiver, fromHex(token, 'bigint'), amount],
+      transactionOverrides,
+    })
+
+    return result
+  }
+
+  protected async _approve({
+    spender,
+    token,
+    amount,
+    transactionOverrides = {},
+  }: WarehouseApproveConfig): Promise<TransactionFormat> {
+    validateAddress(spender)
+    validateAddress(token)
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    const result = await this._executeContractFunction({
+      contractAddress: this._warehouseContract.address,
+      contractAbi: warehouseAbi,
+      functionName: 'approve',
+      functionArgs: [spender, fromHex(token, 'bigint'), amount],
+      transactionOverrides,
+    })
+
+    return result
+  }
+
+  protected async _setOperator({
+    operator,
+    approved,
+    transactionOverrides = {},
+  }: WarehouseSetOperatorConfig): Promise<TransactionFormat> {
+    validateAddress(operator)
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    const result = await this._executeContractFunction({
+      contractAddress: this._warehouseContract.address,
+      contractAbi: warehouseAbi,
+      functionName: 'setOperator',
+      functionArgs: [operator, approved],
+      transactionOverrides,
+    })
+
+    return result
+  }
+
+  protected async _invalidateNonce({
+    nonce,
+    transactionOverrides = {},
+  }: WarehouseInvalidateNonceConfig): Promise<TransactionFormat> {
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    const result = await this._executeContractFunction({
+      contractAddress: this._warehouseContract.address,
+      contractAbi: warehouseAbi,
+      functionName: 'invalidateNonce',
+      functionArgs: [nonce],
+      transactionOverrides,
+    })
+
+    return result
+  }
+
+  protected async _temporaryApproveAndCall({
+    spender,
+    operator,
+    token,
+    amount,
+    target,
+    data,
+    transactionOverrides = {},
+  }: WarehouseTemporaryApproveAndCallConfig): Promise<TransactionFormat> {
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
```

Comment:
nit: keep validating any address inputs?

maybe it's useless to do that. just a pattern I started originally. Was trying to validate all inputs as much as possible so I could give people helpful error messages.

i.e. maybe someone accidentally passes in an ens instead of an address. would want them to get a friendly error message

Reply:
yes, agreed, I was doing this, might have missed it in a few places